### PR TITLE
ddl: wait for 2*lease when finishing delete-reorg in a rollback operation

### PR DESCRIPTION
**Pull request by zimulala**
We encounter the `write conflict` error when doing the delete-range operation. So `TestCancelAddIndex` failed.
When the index in `delete reorg` state we can do the `delete` statement,  and we do the delete-range operation in this state.  So `write conflict` may occur.
This PR fix it.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @winkyao @shenli

File: model/ddl.go [hunk 1]
```diff
@@ -241,7 +246,7 @@ func (job *Job) IsSynced() bool {
 
 // IsDone returns whether job is done.
 func (job *Job) IsDone() bool {
-	return job.State == JobStateDone
+	return job.State == JobStateDone || job.State == JobStateRollbackDone
```
Comment (coocood) [inline model/ddl.go]: The method name `IsDone` seems not consistent with its implementation.
We can keep the original code and use `job.IsDone() || job.IsRollbackDone()` where needed.

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by shenli** [comment]
LGTM

**Pull request merged**